### DISCUSSION TITLE
How to configure legendary?

### DISCUSSION THREAD

**Discussion by danielo515** [LazyVim/LazyVim#2212] [Q&A] [answered]
Can someone share a config of legendary? Mine is not showing any commands. I saw #472 and I already added legendary as a dependency to which-key, but I am still not getting any commands in the legendary prompt.
Here is my config

```lua
return {
  "mrjones2014/legendary.nvim",
  version = "v2.1.0",
  -- since legendary.nvim handles all your keymaps/commands,
  -- its recommended to load legendary.nvim before other plugins
  priority = 10000,
  lazy = false,
  setup = { lazy_nvim = { auto_register = true } },
}

```

**Comment by dpetka2001** [comment, marked as answer]
```lua
  {
    "which-key.nvim",
    dependencies = {
      {
        "mrjones2014/legendary.nvim",
        -- version = "v2.1.0",
        -- since legendary.nvim handles all your keymaps/commands,
        -- its recommended to load legendary.nvim before other plugins
        priority = 10000,
        lazy = false,
        opts = {
          extensions = {
            lazy_nvim = { auto_register = true },
          },
        },
      },
    },
  },
```
This seems to be working. You have to invoke the command `:Legendary` to see the command palette or use a keymap to invoke it.

**Reply by danielo515** [reply]
Yeah, I'm using the legendary command to trigger it, but my options seems to be incorrect. Let me try your fix

**Reply by danielo515** [reply]
The problem was indeed the deps.
Just for completeness, you can still keep your which-key config separated (mine is quite long) and just add the dependency on legendary:

```lua
return {
  "folke/which-key.nvim",
  event = "VeryLazy",
  dependencies = { "legendary.nvim" },
 -- more stuff here
}
```

**Reply by dpetka2001** [reply]
Sure you can. I just did like above to quickly test it in a temporary file. You can have as many fragments of plugins' specs (even in different files) as you want and in the end they all will merge into a final one.

**Reply by danielo515** [reply]
Hey, I just noticed that legendary is not showing any of the already configured which-key commands. Is there a way I can force it? Or do I need to configure it any different?

**Reply by dpetka2001** [reply]
No idea really. I don't use it personally, so I don't know what it does. Maybe better ask at the plugin repo for clarification?

**Reply by dpetka2001** [reply]
https://github.com/mrjones2014/legendary.nvim#configuration
It seems that by default `which-key` is set to `false`, so you would have to enable it I guess.

**Reply by dpetka2001** [reply]
add 
```lua
            which_key = {
              auto_register = true,
            },
```
inside extensions like `lazy_nvim`

**Reply by danielo515** [reply]
True, thank you very much. I thought that was a default, but even with that, not a single map appears.

```lua
{
  "mrjones2014/legendary.nvim",
  version = "v2.1.0",
  -- since legendary.nvim handles all your keymaps/commands,
  -- its recommended to load legendary.nvim before other plugins
  priority = 10000,
  lazy = false,
  opts = {
    lazy_nvim = { auto_register = true },
    which_key = { auto_register = true },
    extensions = {
      nvim_tree = true,
      smart_splits = false,
      op_nvim = false,
      diffview = false,
    },
  },
}
```

And this is my which-key config
```lua
{
  "folke/which-key.nvim",
  event = "VeryLazy",
  dependencies = { "legendary.nvim" },
  opts = {
    plugins = { spelling = true },
    defaults = {
      ["<leader>w"] = { name = "Save all" },
      ["<leader>ww"] = false,
      ["<leader>wd"] = false,
      ["<leader>w-"] = false,
      ["<leader>w|"] = false,
    },
  },
  config = function(_, opts)
    local wk = require("which-key")
    opts.defaults["<leader><Space>"] = nil
    opts.defaults["<leader><tab>"] = nil
    opts.defaults["<leader>w"] = nil
    opts.defaults["<leader>gh"] = nil
    -- wk.register(opts.defaults)
    wk.setup(opts)
    wk.register(whichMaps, { prefix = "<leader>" })
  end,
}
```

**Reply by dpetka2001** [reply]
They have to be inside the `extensions` table like i did in my first post.

**Reply by danielo515** [reply]
I was just following the docs, where they are clearly two separated options:
```lua
  lazy_nvim = {
    -- Automatically register keymaps that are defined on lazy.nvim plugin specs
    -- using the `keys = {}` property.
    auto_register = false,
  },
  which_key = {
    -- Automatically add which-key tables to legendary
    -- see ./doc/WHICH_KEY.md for more details
    auto_register = false,
    -- you can put which-key.nvim tables here,
    -- or alternatively have them auto-register,
    -- see ./doc/WHICH_KEY.md
    mappings = {},
    opts = {},
    -- controls whether legendary.nvim actually binds they keymaps,
    -- or if you want to let which-key.nvim handle the bindings.
    -- if not passed, true by default
    do_binding = true,
    -- controls whether to use legendary.nvim item groups
    -- matching your which-key.nvim groups; if false, all keymaps
    -- are added at toplevel instead of in a group.
    use_groups = true,
  },
  -- Which extensions to load; no extensions are loaded by default.
  -- Setting the plugin name to `false` disables loading the extension.
  -- Setting it to any other value will attempt to load the extension,
  -- and pass the value as an argument to the extension, which should
  -- be a single function. Extensions are modules under `legendary.extensions.*`
  -- which return a single function, which is responsible for loading and
  -- initializing the extension.
  extensions = {
    nvim_tree = false,
    smart_splits = false,
    op_nvim = false,
    diffview = false,
  },
```

Gonna give it a try anyway

**Reply by dpetka2001** [reply]
I know, but when I tried it I got a warning about being deprecated and they should be moved to extensions instead. See [here](https://github.com/mrjones2014/legendary.nvim/blob/234e2ef8dec65db153c63a8b310c74155bb93ee7/lua/legendary/deprecate.lua#L61-L76).

**Comment by danielo515** [comment]
Lovely!

On Mon, Dec 18, 2023 at 2:43 PM Iordanis Petkakis ***@***.***>
wrote:

> Sure you can. I just did like above to quickly test it in a temporary
> file. You can have as many fragments of plugins' specs (even in different
> files) as you want and in the end they all will merge into a final one.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/LazyVim/LazyVim/discussions/2212#discussioncomment-7886664>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARKJWPODJ4K75QDDGVRP4TYKBB7DAVCNFSM6AAAAABAZPYMKKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TQOBWGY3DI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
---
https://danielorodriguez.com
